unittests/ASM/CMakeLists.txt

```diff
@@ -90,6 +90,8 @@ foreach(ASM_SRC ${ASM_SOURCES})
       list(APPEND ARGS_LIST "--smcchecks=full")
     endif()
 
+    list(APPEND ARGS_LIST "--")
```

Comment:
Why is this needed now when it wasn't before?

Reply:
We had a bug in our arguments getting passed to unittests for years (#4071) and it was undiscovered. Due to a quirk with cpp-optparse where any unsupported arguments were gathered and passed to the guest application. This didn't necessarily break anything in our unittests, just made them run silently, and for all the harness things the additional argument was ignored anyway.

The commit these changes are from should probably be broken out to a freestanding commit before this PR is merged anyway.

Reply:
and since I realize I didn't actually answer the question. We're being more strict than cpp-optparse splitting out FEXLoader arguments from guest arguments because of that mentioned problem. So the unittests need the split split pattern now, where cpp-optparse didn't need this.

Reply:
> We're being more strict than cpp-optparse splitting out FEXLoader arguments from guest arguments because of that mentioned problem. So the unittests need the split pattern now, where cpp-optparse didn't need this.

Sounds reasonable 👍